# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1546550515 -3600
#      Thu Jan 03 22:21:55 2019 +0100
# Branch stable
# Node ID 25dfa8c96b7b2f241f07cb53dc14277c1fed2d95
# Parent  31b0761cd641ff6038818c0ab45000d3ba3d2149
* documentation.cc (notice_settings): prevent segfault if help engine does not exist

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -533,16 +533,21 @@ namespace octave
 
       // Apply selection and move back to the beginning
       m_doc_browser->setExtraSelections (selected);
       m_doc_browser->moveCursor (QTextCursor::Start);
   }
 
   void documentation::notice_settings (const QSettings *settings)
   {
+    // If m_help_engine is not defined, the object accessed by this method
+    // are not valid. Thus, just return in this case
+    if (! m_help_engine)
+      return;
+
     // Icon size in the toolbar.
     int size_idx = settings->value (global_icon_size.key,
                                     global_icon_size.def).toInt ();
     size_idx = (size_idx > 0) - (size_idx < 0) + 1;  // Make valid index from 0 to 2
 
     QStyle *st = style ();
     int icon_size = st->pixelMetric (global_icon_sizes[size_idx]);
     m_tool_bar->setIconSize (QSize (icon_size, icon_size));
